### PR TITLE
PR for #3718: Remove CommanderCacher class and related code

### PR DESCRIPTION
**Pull request by edreamleo**
See #3718.

**Overview**

The following relationships *remain unchanged*:

- `g.app.db` is a singleton instance of the `GlobalCacher` class.
-  For each commander, `c.db` is an instance of the `CommanderWrapper` class.
   However, Leo creates `c.db` without using the (deleted) `CommanderCacher` class.

The `.leo/db` folder now contains only a `g_app_db` folder.
Leo no longer writes per-commander sub-folders in the  `.leo/db` folder!

**Details**

- [x] Delete the `CommanderCacher` class.
- [x] Instantiate per-commander `CommanderWrapper` instances in `c.initObjects`.
- [x] Remove all references to the `g.app.commander_cacher` and `g.app.commander_db` ivars.
- [x] Fix error recover code in `GlobalCacher.clear`.
- [x] Remove mypy suppression in `GlobalCacher.clear`.

**Extras**

- [x] Remove the never-used `fn` kwarg to `CommanderWrapper.__init__`.
- [x] `app.finishQuit` no longer calls `pyzo_close_handler`.

Later (in a separate PR) or never: `clear-all-caches` deletes entire contents of `./leo/db'.

### CODE REVIEW

**Comment by edreamleo** [comment]
@boltex I'm fuzzy about why the calls to ` g.app.commander_cacher.save` aren't needed in `leoFileCommands.py`.

Everything seems to work, but I don't understand why? Could you explain?

**Comment by edreamleo** [comment]
@boltex Oh. I think I understand. There is only one db. The wrapper simply creates commander-specific keys, as you mentioned in #3718. So the save commands need not save anything. Correct?

**Comment by boltex** [comment]
@edreamleo  In Leo, the db is opened in auto-commit mode. this may explain why saves and commits are unnecessary.

`
        self.conn = sqlite3.connect(dbfile, isolation_level=None)
`

The default autocommit is LEGACY_TRANSACTION_CONTROL, which makes the isolation_level=None setting enforce the 'autocommit' behavior. **This is as per: https://docs.python.org/3/library/sqlite3.html#sqlite3.Connection.autocommit**   

notice the _recommended_ VS _default_ value of 'autocommit' which I had to re-read 3 times to make sure this is all true regarding isolation_level having the final word !

(I had found that out while studying the implementation of the sqlite used by Leo for this. and the options it uses when using the python sqlite library.)

Also, to confirm everything you can peek very easily into the db at crucial points in time while using Leo with https://sqlitebrowser.org/ to make sure stuff is properly set in the db when you think it should.

**Comment by boltex** [comment]
I'll test and review this major fix soon.

**Comment by edreamleo** [comment]
@boltex Thanks for your comments. I had no idea the db settings were important.

File: leo/core/leoCache.py [hunk 1]
```diff
@@ -143,9 +65,16 @@ def __getitem__(self, key: str) -> Any:
     def __setitem__(self, key: str, value: Any) -> None:
         self.user_keys.add(key)
         self.db[f"{self.key}:::{key}"] = value
-#@+node:ekr.20180627041556.1: ** class GlobalCacher
+#@+node:ekr.20180627041556.1: ** class GlobalCacher (g.app.db)
 class GlobalCacher:
-    """A singleton global cacher, g.app.db"""
+    """
+    A calss creating a singleton global database, g.app.db.
```
Comment (boltex) [inline leo/core/leoCache.py:71]: class instead of calss

**Review by boltex — APPROVED** [review]
Looks Good! Can't wait to also clean that up completely in leojs!!
I also tried it with sessions and expanded/marked node changes, etc.. Everything seems to work just fine 👍

File: leo/core/leoCache.py (hunk 1 above)
Comment (edreamleo) [reply leo/core/leoCache.py]: Typo fixed at 5199444.

**Comment by edreamleo** [comment]
@boltex Thanks for your comments and testing. This PR is a big step forward.

**Pull request merged**